### PR TITLE
[PT FE]: support aten::log1p, fixes for where and linalg_norm

### PR DESCRIPTION
**Pull request by eaidova**
### Details:
 - *support aten::log1p*
 - *extended linal_norm for tensors with rang > 2 if ord not provided*
 - *fixed aten::where translation for mixed precision if/else tensors*
 
verified with owl-vit models family

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/frontends/pytorch/src/op/norm.cpp [hunk 1]
```diff
@@ -259,11 +259,8 @@ OutputVector translate_linalg_norm(const NodeContext& context) {
             auto input_rank = x.get_partial_shape().rank();
             if (input_rank.is_static() && input_rank.get_length() == 2) {
                 result = frobenius_norm(context, x, dim, keep_dim);
-            } else if (input_rank.is_static() && input_rank.get_length() == 1) {
-                result = norm_vector(context, x, dim, 2, keep_dim);
             } else {
-                FRONT_END_OP_CONVERSION_CHECK(false,
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/norm.cpp:265]: Does it change the behaivior or just allows dynamic rank? If the later I think it makes sense to reverse the validation. Previously it was working for `input_rank.is_static() && input_rank.get_length() == 1`. Replace it with working if `input_rank.is_dynamic() || input_rank.get_length() == 1`

File: tests/layer_tests/pytorch_tests/test_log.py [hunk 2]
```diff
@@ -42,7 +44,9 @@ def forward(self, x):
                              ["log_", "float32"], 
                              ["log2", "float32"], 
                              ["log2", "int32"], 
-                             ["log2_", "float32"]])
+                             ["log2_", "float32"],
+                             ["log1p", "int32"],
```
Comment (rkazants) [inline tests/layer_tests/pytorch_tests/test_log.py:48]: Do we need to test `log1p` for float32?

File: src/frontends/pytorch/src/op/norm.cpp (hunk 1 above)
Comment (eaidova) [reply src/frontends/pytorch/src/op/norm.cpp:265]: I'm not sure that it is my case, I'll check. I thought that in my case rank known but it is greater 2

Comment (eaidova) [reply src/frontends/pytorch/src/op/norm.cpp:265]: yes, it works in my case, updated

File: tests/layer_tests/pytorch_tests/test_log.py (hunk 2 above)
Comment (eaidova) [reply tests/layer_tests/pytorch_tests/test_log.py:49]: added, thanks

**Review by mvafin — APPROVED** [review]

**Pull request merged**